lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/IMRUDriver.cs

```diff
@@ -161,6 +169,12 @@ internal sealed class IMRUDriver<TMapInput, TMapOutput, TResult, TPartitionType>
             _serviceAndContextConfigurationProvider =
                 new ServiceAndContextConfigurationProvider<TMapInput, TMapOutput, TPartitionType>(dataSet, configurationManager);
 
+            if (lifecycleManager != null)
+            {
+                var handle = lifecycleManager.Subscribe(this as IObserver<IJobCancelled>);
+                _disposableResources.Add(handle);
```

Comment:
Not easily, there is some logic in manager which depends on timer lifecycle to be the same as manager lifecycle. We do dispose timer when JobLifecyclemanager is destructed. that should be enough. We can change the design to separate Timer lifecycle from JobManager lifecycle....let me know if feel strongly about it.